docs/src/main/asciidoc/_migration-guide-for-4.0.adoc

```diff
@@ -1224,6 +1227,81 @@ spring:
                   inclusive: false
 ----
 
+If you use security principals instead of connection strings, in versions before 4.0 the application will firstly connect to Azure Resource Manager (ARM) with the provided security principal, and then
+retrieve the connection string of the specified namespace with ARM. In the end the application uses the retrieved connection string to connect to Azure Event Hubs. In this way
+the provided security principal should be granted with the link:https://docs.microsoft.com/azure/role-based-access-control/built-in-roles#contributor[Contributor] role to retrieve of the associated Azure Event Hubs namespace.
+
+For Azure Spring Cloud 4.0, we provide two ways of leveraging security principals for authentication. One is still using the principals to connect to ARM and retrieve the connection strings where the `Contributor` role is required for the principals.
+The other leverages security principals to authenticate to Azure Active Directory (Azure AD) and then connect to Azure Event Hubs directly, in this case the `Contributor` role is not necessary anymore
+while other `Data` related roles are required for messaging operations, please refer to link:index.html#authorize-access-with-azure-active-directory[Authorize access with Azure AD] to make sure the security principal has been granted the sufficient permission to access the Azure resource.
+
+For authentication based on ARM, taking service principal as example, configuration migration is listed as below, where the assigned role should not change:
+
+[source,yaml]
+----
+spring:
+  cloud:
+    azure:
+      client-id: ${CLIENT_ID}
+      client-secret: ${CLIENT_SECRET}
+      tenant-id: ${TENANT_ID}
+      resource-group: ${EVENTHUB_RESOURCE_GROUP}
+      eventhub:
+        namespace: ${EVENTHUB_NAMESPACE}
+----
+
+Modern configuration, properties for Azure subscription id and resource group are required:
+
+[source,yaml]
+----
+spring:
+  cloud:
+    azure:
+      credential:
+        client-id: ${CLIENT_ID}
+        client-secret: ${CLIENT_SECRET}
+      profile:
+        tenant-id: ${TENANT_ID}
+        subscription-id: ${SUBSCRIPTION_ID}
+      eventhubs:
+        namespace: ${EVENTHUB_NAMESPACE}
+        resource:
+          resource-group: ${RESOURCE_GROUP}
+----
+
+You can also migrate to authenticate and authorize with Azure AD directly without making a detour to ARM. Security principals of the service principal and the manged identity are both supported, where the principal should
+be granted with necessary `Data` related roles for messaging operations. The configuration examples of the service principal and the manged identity are listed as below:
```

Comment:
How about changing this `Security principals of the service principal and the manged identity are both supported, where the principal should
be granted with necessary `Data` related roles for messaging operations.` to `Make sure to grant the security principal necessary `Data` roles for messaging operations.` directly. The former expression will make people think we only support service principal and mi, but we do support user principals.

Reply:
good idea, I ignored user principals as well